src/EditorFeatures/Core/Wrapping/BinaryExpression/AbstractBinaryExpressionWrapper.cs

```diff
@@ -0,0 +1,100 @@
+        {
+            if (!(node is TBinaryExpressionSyntax binaryExpr))
+            {
+                return default;
+            }
+
+            if (!IsLogicalExpression(binaryExpr))
+            {
+                return default;
+            }
+
+            // Don't process this binary expression if it's in a parent logical expr.  We'll just
+            // allow our caller to walk up to that and call back into us to handle.  This way, we're
+            // always starting at the topmost logical binary expr.
+            if (IsLogicalExpression(binaryExpr.Parent))
+            {
+                return default;
+            }
+
+            var exprsAndOperators = GetExpressionsAndOperators(binaryExpr);
+#if DEBUG
+            Debug.Assert(exprsAndOperators.Length >= 3);
+            Debug.Assert(exprsAndOperators.Length % 2 == 1, "Should have odd number of exprs and operators");
+            for (int i = 0; i < exprsAndOperators.Length; i++)
+            {
+                var item = exprsAndOperators[i];
+                Debug.Assert(((i % 2) == 0 && item.IsNode) ||
+                             ((i % 2) == 1 && item.IsToken));
+            }
+#endif
+
+            var containsUnformattableContent = await ContainsUnformattableContentAsync(
+                document, exprsAndOperators, cancellationToken).ConfigureAwait(false);
+
+            if (containsUnformattableContent)
+            {
+                return default;
+            }
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            return new BinaryExpressionCodeActionComputer(
+                this, document, sourceText, options, binaryExpr,
+                exprsAndOperators, cancellationToken);
+        }
+
+        private ImmutableArray<SyntaxNodeOrToken> GetExpressionsAndOperators(TBinaryExpressionSyntax binaryExpr)
+        {
+            var result = ArrayBuilder<SyntaxNodeOrToken>.GetInstance();
+            AddExpressionsAndOperators(binaryExpr, result);
+            return result.ToImmutableAndFree();
+        }
+
+        private void AddExpressionsAndOperators(
+            SyntaxNode expr, ArrayBuilder<SyntaxNodeOrToken> result)
+        {
+            if (IsLogicalExpression(expr))
+            {
+                _syntaxFacts.GetPartsOfBinaryExpression(
+                    expr, out var left, out var opToken, out var right);
+                AddExpressionsAndOperators(left, result);
+                result.Add(opToken);
+                AddExpressionsAndOperators(right, result);
+            }
+            else
+            {
+                result.Add(expr);
+            }
+        }
+
+        private bool IsLogicalExpression(SyntaxNode node)
+            => _syntaxFacts.IsLogicalAndExpression(node) ||
+               _syntaxFacts.IsLogicalOrExpression(node);
```

Comment:
❔ Any reason to not include XOR?

Reply:
It wasn't explicitly avoided, it just seemed like the primary cases that appear in general are long sequences of &&'s and ||'s.  If you had an ^ in it, it seems more likely that would be part of some arithmetic expression you would *not* want wrapped.

Reply:
On that note, it seems like we _would_ want to support `+`, even if `^` and/or `-` were omitted.